app/core/modals/SupportingMemberSignupModal.tsx

```diff
@@ -168,23 +172,23 @@ export default function SignupModal({ button }) {
                       type="button"
                       onClick={() =>
                         handlePayment(
-                          `/api/checkout_sessions_supporting?email=${encodeURIComponent(
-                            currentUser?.email!
-                          )}&price_id=${price}`,
+                          `/api/checkout_sessions_supporting${
+                            sponsored ? "_sponsored" : ""
+                          }?email=${encodeURIComponent(currentUser?.email!)}&price_id=${price}`,
```

Comment:
This may be an out-of-scope in this PR, but I'd think it would be a good practice to work on types so that we can avoid using the bang (`!`).